src/fixLockfileIntegrity.ts

```diff
@@ -88,15 +89,16 @@ export const fixLockFile = async (lockFileLocation: string): Promise<FixLockFile
     let promises: Array<Promise<void>> = [];
     let hashesFound: Set<string> = new Set<string>();
     traverse(lockFile).forEach(function(node) {
-        if (node && node.version && node.resolved && new URL(node.resolved)?.host === REGISTRY_HOST && node.integrity?.startsWith("sha1-")) {
-            const packageName = parsePackageName(this.key);
-            const packageVersion = node.version;
-            const oldIntegrity = node.integrity;
+        if (node && node.version && node.resolved && node.integrity?.startsWith("sha1-") && isRegistrySupported(new URL(node.resolved)?.host)) {
+            const packageName: string = parsePackageName(this.key);
+            const registry: string = node.resolved.substring(0, node.resolved.indexOf(packageName));

```

Comment:
```suggestion
            const registry: string = node.resolved.substring(0, node.resolved.indexOf("/" + packageName + "/") + 1);
```
Quick bugfix - otherwise registry inference will fail for simply named packages such as [this](https://www.npmjs.com/package/i).